arch/arm/src/common/arm_interruptcontext.c

```diff
@@ -59,5 +59,15 @@
 
 bool up_interrupt_context(void)
 {
-  return CURRENT_REGS != NULL;
+#ifdef CONFIG_SMP
```

Comment:
```suggestion
#ifdef CONFIG_SMP
  /* REVISIT:  Currently up_irq_save() will not disable the Software generated interrupts (SGIs)
   * for the case of ARMv7-A architecture using the GIC.  So this will not be sufficient in
   * that case, at least not until we add support for the ICCMPR.
   */

```

Reply:
I tried to commit this so that I could merge your changes with this comment, but the lines are too long.  I will try to modify you fork to fix the line width.